api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -26,6 +26,26 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 //    without `case normalization <https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.1>`_
 //
 message HeaderValidatorConfig {
+  // Action to take when Envoy receives client request with header names containing underscore
+  // characters.
+  // Underscore character is allowed in header names by the RFC-7230 and this behavior is implemented
+  // as a security measure due to systems that treat '_' and '-' as interchangeable. Envoy by default allows client request headers with underscore
+  // characters.
+  enum HeadersWithUnderscoresAction {
+    // Allow headers with underscores. This is the default behavior.
+    ALLOW = 0;
+
+    // Reject client request. HTTP/1 requests are rejected with the 400 status. HTTP/2 requests
+    // end with the stream reset. The "httpN.requests_rejected_with_underscores_in_headers" counter
+    // is incremented for each rejected request.
+    REJECT_REQUEST = 1;
+
+    // Drop the client header with name containing underscores. The header is dropped before the filter chain is
+    // invoked and as such filters will not see dropped headers. The
+    // "httpN.dropped_headers_with_underscores" is incremented for each dropped header.
```

Comment:
The "httpN.dropped_headers_with_underscores" should be a reference to the doc containing the stat.
(This PR should also have some docs)

Reply:
Docs were added as part of #21172. This just added a new field

Reply:
nit: This should be a reference to the `_config_http_conn_man_stats_per_codec` section (in `docs/root/configuration/http/http_conn_man/stats.rst`).

Reply:
I added a reference to the counter.